### ISSUE TITLE
Alerts for dashboards don't create alerts on grafana (+ Add alerts to example dashboards)

### ISSUE THREAD

**Issue by ArzelaAscoIi** [weaveworks/grafanalib#529]
I am trying to add alerts to my graphs and added Alerts according to the documentation (and checking the code), but these alerts are not reflected in Grafana. I tried multiple configurations for the alerts but none of them got reflected in grafana.

Here are my examples:

## AlertCondition list for alerts
```py
panels.append(
    Graph(
        title="Random Walk",
        dataSource={"type": "loki", "uid": "grafanacloud-logs"},
        targets=[
            Target(
                datasource={"type": "loki", "uid": "grafanacloud-logs"},
                expr='sum by (json_field) (count_over_time({...} |="Filter" | json | __error__=""  [5m]))',
            ),
        ],
        gridPos=GridPos(h=8, w=16, x=0, y=0),
        alert=[  # <-- the docstring of the constructur of the Graph class says that this should be a list of alert conditions
            AlertCondition( 
                target=Target(
                    datasource={"type": "loki", "uid": "grafanacloud-logs"},
                    expr='sum by (json_field) (count_over_time({...} |="Filter" | json | __error__=""  [5m]))',
                ),
                evaluator=Evaluator(type=EVAL_GT, params=42),
                timeRange=TimeRange(from_time="5m", to_time="now"),
                operator=OP_AND,
                reducerType=RTYPE_AVG,
            )
        ],
    )
)
```

## List of alerts
```py
Graph(
        title="Random Walk",
        dataSource={"type": "loki", "uid": "grafanacloud-logs"},
        targets=[
            Target(
                datasource={"type": "loki", "uid": "grafanacloud-logs"},
                expr='sum by (json_field) (count_over_time({...} |="Filter" | json | __error__=""  [5m]))',
            ),
        ],
        gridPos=GridPos(h=8, w=16, x=0, y=0),
        alert=[
            Alert(
                name="My alert",
                message="My alert message",
                noDataState="OK",
                alertConditions=[
                    AlertCondition(
                        target=Target(
                            datasource={"type": "loki", "uid": "grafanacloud-logs"},
                            expr='sum by (json_field) (count_over_time({...} |="Filter" | json | __error__=""  [5m]))',
                        ),
                        evaluator=Evaluator(type=EVAL_GT, params=42),
                        timeRange=TimeRange(from_time="5m", to_time="now"),
                        operator=OP_AND,
                        reducerType=RTYPE_AVG,
                    )
                ],
            )
        ],
    ) 
```

## What happened?
- no error messages 

## How to reproduce it?
Both of these examples wont produce alerts on grafana. 

# Idea
I think it would be super helpful to have an example of an alert in the example dashboards, s.t. others don't face the same issue I have. Happy to contribute once this is fixed :)

**Comment by JamesGibo** [comment]
Which version of Grafana are you using? 
Alerting has changed in Grafana and the latest version do not support this method.
See #523 for more information

**Comment by ArzelaAscoIi** [comment]
Thanks @JamesGibo for your reply. I am using grafana cloud. Your were right about the incompatibility. I just tried the latest changes from @OscarVanL and it works like a charm 🎉 

I only had to adjust the datasource to the name i was using [here](https://github.com/weaveworks/grafanalib/blob/b547a0bc879cf46fbd6bd67b66115509c0dda513/grafanalib/tests/examples/example.alertgroup.py#L42). A comment here might be helpful, but this is nitpicking :) 

Thanks!
